qiime/workflow/pick_open_reference_otus.py

```diff
@@ -1091,3 +1091,225 @@ def pick_subsampled_open_reference_otus(input_fp,
     if not suppress_index_page:
         index_fp = '%s/index.html' % output_dir
         generate_index_page(index_links, index_fp)
+
+
+def convergent_pick_subsampled_open_reference_otus(
+        input_fps,
+        refseqs_fp,
+        output_dir,
+        percent_subsample,
+        new_ref_set_id,
+        command_handler,
+        params,
+        qiime_config,
+        prefilter_refseqs_fp=None,
+        prefilter_percent_id=None,
+        min_otu_size=2,
+        run_assign_tax=True,
+        run_align_and_tree=True,
+        step1_otu_map_fp=None,
+        step1_failures_fasta_fp=None,
+        parallel=False,
+        suppress_step4=False,
+        logger=None,
+        suppress_md5=False,
+        denovo_otu_picking_method='uclust',
+        reference_otu_picking_method='uclust_ref',
+        status_update_callback=print_to_stdout,
+        minimum_failure_threshold=100000,
+        num_seqs=10000):
+    """Call the convergent open reference workflow"""
+    create_dir(output_dir)
+    commands = []
+
+    if logger is None:
+        logger = WorkflowLogger(generate_log_fp(output_dir), params=params,
+                                qiime_config=qiime_config)
+        close_logger_on_success = True
+    else:
+        close_logger_on_success = False
+
+    # if the user has not passed a different reference collection for the
+    # pre-filter, used the input refseqs_fp for all iterations. we want to
+    # pre-filter all data against the input data as lower percent identity
+    # searches with uclust can be slow, so we want the reference collection to
+    # stay at a reasonable size.
+    if prefilter_refseqs_fp is None:
+        prefilter_refseqs_fp = refseqs_fp
+
+    otu_table_fps = []
+    repset_fasta_fps = []
+
+    # Construct an array of sequence generators
+    seq_generators = [parse_fasta(open(fp, 'U')) for fp in input_fps]
+
+    done = False
+    iteration = 0
+
+    # Iterate until all seqs are clustered
+    while not done:
+        iteration_output_dir = '%s/%d/' % (output_dir, iteration)
+        iter_input_fp = join(output_dir, "iter_%s_seqs.fna" % iteration)
+
+        next_iter_seq_gen = []
+
+        if len(seq_generators) == 0:
+            done = True
```

Comment:
I think this would be clearer if you did this as ``while len(seq_generators) > 0``, rather than creating a separate ``done`` variable. Would that work?